core/src/main/java/org/apache/carbondata/core/metadata/schema/table/CarbonTable.java

```diff
@@ -1165,4 +1165,30 @@ private static void setLocalDictInfo(CarbonTable table, TableInfo tableInfo) {
       table.setLocalDictionaryEnabled(Boolean.parseBoolean("false"));
     }
   }
+
+  /**
+   * Method to get the list of cached columns of the table
+   *
+   * @param tableName
+   * @return
+   */
+  public List<String> getCachedColumns(String tableName) {
+    List<String> cachedColsList = new ArrayList<>(tableDimensionsMap.size());
+    String cacheColumns =
+        tableInfo.getFactTable().getTableProperties().get(CarbonCommonConstants.COLUMN_META_CACHE);
+    List<CarbonDimension> carbonDimensions = tableDimensionsMap.get(tableName);
```

Comment:
Check in complete columns not just dimensions

Reply:
Ok